crnk-core/src/main/java/io/crnk/core/boot/CrnkBoot.java

```diff
@@ -66,7 +66,7 @@
 
 	private QueryParamsBuilder queryParamsBuilder;
 
-	private QuerySpecDeserializer querySpecDeserializer = new DefaultQuerySpecDeserializer();
+	private QuerySpecDeserializer querySpecDeserializer;
```

Comment:
set currently breaks with the the setter of default/max limit, where did you run into issues? should be ok to have the default one?

Reply:
That's actually was my main problem. I wanted to oeverride the `QuerySpecDeserializer ` impl and couldn't because of this explicit definition. And that's the main intention of this PR actually.

Reply:
with the updated spring config in this PR that should not be an issue anymore, even if there is still default instance that will then be overridden?

Reply:
I don't understand your question.. currently `QuerySpecDeserializer` is defined explicitly so the only way to override it is to override the whole CrnkBoot class. There is a setter though but it's useless anyway because of `checkNotConfiguredYet()` check. What I wanted to implement is the dynamic instance creation (DI). This way anyone who wants his version of deserializer will just define a bean of appropriate type, default one will be used as a fallback mechanism.

Reply:
I will take a closer look the next two, three days (today already occupied)